src/render_backend.rs

```diff
@@ -79,27 +85,45 @@ impl RenderBackend {
             webrender_context_handle: webrender_context_handle,
             webgl_contexts: HashMap::new(),
             current_bound_webgl_context_id: None,
-        }
+            enable_recording: enable_recording,
+	}
     }
 
     pub fn run(&mut self) {
         let mut profile_counters = BackendProfileCounters::new();
-
         loop {
             let msg = self.api_rx.recv();
             match msg {
```

Comment:
I think it might be clearer if you did something like this above this line:

```rust
if self.enable_recording {
    let mut file = OpenOptions::new().append(true).create(true).open("resources.bin").unwrap();
    let buff = serialize(&msg, bincode::SizeLimit::Infinite).unwrap();
    write_msg(&mut file, &buff);
}
```

That should remove a lot of code duplication. Also, I'd consider opening the file just once above the loop, and not using `unwrap()` so heavily, but handling the error. That being said, given it's behind an option, maybe it's not such a big deal :)